api.proto

```diff
@@ -58,6 +58,7 @@ message SubscribeRequest {
     StartPosition startPosition  = 3; // Where to begin consuming from
     int64         startOffset    = 4 [jstype=JS_STRING]; // Offset to begin consuming from
     int64         startTimestamp = 5 [jstype=JS_STRING]; // Timestamp to begin consuming from
+    bool          readReplica    = 6;  // Subscribe from Replica instead of leader
```

Comment:
We ought to think about the semantics of this a bit. I think rather than trying to _guarantee_ a replica will be selected, we just select a random member from the ISR (any of the followers or leader). I think the use case here is we want to balance load amongst ISR members, so I think we should include the leader in that. Moreover, there's actually a bit of additional work needed to guarantee the server receiving the request is, in fact, a follower.

Reply:
I hesitated in the beginning between these 2.  In this case, I think your proposition makes sense as by selecting a random ISR replica, the purpose of fan-out has been achieved without further complication. Change has been made.

But I am curious to know, what additional work you mention to guarantee the server receiving the request ?